Integration work  as we bring up the swerve robot functionality

bot190 opened the pull request:
Adds command mappings for basic functionality.
Some improvements in the other subsystems as well.

bot190 committed:
Update Spindexer Gear Ratio

Changes to the gearing resulted in a slight change of the overall gear ratio.

msoucy committed:
Imports

bot190 committed:
Integration work

Add command mappings for initial functionality
Add enum to track preset shooter speeds and simplify command structure.

bot190 committed:
Integration work

Get Intake and Spindexer working together correctly
Map kicker to POV Down for testing

jeffaburke committed:
Updated for PR comments

Added a Direction enum for turret

jeffaburke committed:
Updated for PR comments

jeffaburke committed:
Merge remote-tracking branch 'origin/Integration' into Integration

bot190 committed:
Final Integration before RiverRage

bot190 committed:
RiverRage Changes

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -45,20 +52,59 @@
+        // Shooter mappings
+        driveController.getButton(Button.kB).whileHeld(shootPowerCell(shooterSpeeds.GoalBase))
+                .whenReleased(parallel("Stop", shooter.spinDown(), kicker.stopKicker()));
+        // driveController.getButton(Button.kX).whileHeld(shootPowerCell(shooterSpeeds.TrenchShot))
```
Commented out code

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -45,20 +52,59 @@
+        driveController.getButton(Button.kBumperLeft).whileHeld(turret.slowRotate(Turret.Direction.COUNTERCLOCKWISE));
+
+        // Secondary functions mapped to the dpad
+        driveController.getPovButton(Direction.Left).whileHeld(intake.runIntake(false));
```
This and others - I hate boolean parameters, they don't really explain what they do.

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -72,7 +118,11 @@ public void setDefaultCommands() {
     @Override
     public void populateAutonomous() {
-        // TODO: Define autonomous
+        // autoChooser.setDefaultOption("Initialize Systems", new
```
Commented code

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -72,7 +118,11 @@ public void setDefaultCommands() {
+        // SequentialCommandGroup(intake.deployIntake()));
+        autoChooser.setDefaultOption("Do Nothing", instant("Do nothing", () -> {
+        }));
+        autoChooser.addOption("Driveoff", drive.driveDistanceY(5));
```
Should use `addAutonomous`, but it's missing a default option... so not until it's fixed

msoucy commented on src/main/java/frc/robot/maps/SwerveBot.java:
```diff
@@ -171,8 +178,8 @@ public TurretMap getTurretMap() {
-        // return new TurretMap(motor, limitSwitch);
-        return new TurretMap();
+        return new TurretMap(motor, limitSwitch);
+        // return new TurretMap();
```
Commented code

msoucy commented on src/main/java/frc/robot/maps/SwerveBot.java:
```diff
@@ -194,48 +201,49 @@ public ShooterMap getShooterMap() {
+        // hoodPID.setD(0);
+
+        return new ShooterMap(shooterA, new SmartMotorController(), new SmartMotorController());
+        // return new ShooterMap();
```
Does our shooter not shoot?

msoucy commented on src/main/java/frc/robot/subsystems/Kicker.java:
```diff
@@ -39,14 +47,26 @@ public CommandBase KickToShoot(int numBalls) {
     }
 
+    public CommandBase startKicker() {
+        return new SequentialCommandGroup(new InstantCommand(() -> {
```
You can use `CommandRobot.sequence` to avoid needing `withName`, and `instant` instead of `new InstantCommand`

bot190 replied:
Created chopshop-166/chopshoplib#79 to add `sequence` and `parallel` to `SmartSubsystem` since they're missing for some reason.

msoucy commented on src/main/java/frc/robot/subsystems/Kicker.java:
```diff
@@ -39,14 +47,26 @@ public CommandBase KickToShoot(int numBalls) {
 
     // Fallback command if the sensor isn't working
-    public CommandBase runKicker() {
+    public CommandBase runKicker(boolean up) {
```
Smurf naming

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -37,25 +35,43 @@ public Shooter(ShooterMap map) {
         // The roller check is less restrictive as it should have less impact
-        final PersistenceCheck rollerCheck = new PersistenceCheck(5,
-                () -> Math.abs(rollerEncoder.getRate() - ROLLER_SPEED) <= 200.0);
+        // final PersistenceCheck rollerCheck = new PersistenceCheck(5,
```
Commented out stuff

msoucy commented on src/main/java/frc/robot/subsystems/Spindexer.java:
```diff
@@ -51,13 +56,20 @@ public CommandBase spinDistance(final double distance) {
+
+    @Override
+    public void periodic() {
+        // TODO Auto-generated method stub
```
`TODO`?

bot190 replied:
`TODONE`?

msoucy commented on src/main/java/frc/robot/subsystems/Turret.java:
```diff
@@ -88,37 +93,45 @@ public CommandBase aimForward() {
             motor.set(0);
         }, () -> {
-            return Math.abs(angle - encoder.getAbsolutePosition()) <= POSITION_ERROR;
+            return Math.abs(angle - encoder.getDistance()) <= POSITION_ERROR;
```
Get rid of the `return` and curly braces and it'll look prettier imo

bot190 committed:
PMD clean-up

bot190 committed:
Code clean-up

Resolves most of the PR comments.

bot190 committed:
Battle of the Governor's Cup changes

Various changes made at Battle of the Bay to improve robot functionality.

bot190 committed:
Update to latest PhotonLib vendordep

PhotonLib released a new version and changed their file naming that broke the build. Pull in the latest json file so we can build with PhotonLib.
